src/client.ts

```diff
@@ -75,17 +75,21 @@ export class WaClient {
     }
 
     createConnection = () => {
-        this.socket = new Socket();
-        this.socket.open();
-        this.ephemeralKeyPair = generateIdentityKeyPair();
-        this.socket.onOpen = this.handleSocketOpen;
+        return new Promise((resolve, reject) => {
+            this.socket = new Socket();
+            this.socket.open();
+            this.ephemeralKeyPair = generateIdentityKeyPair();
+            this.socket.onOpen = this.handleSocketOpen(resolve, reject);
+        });
     };
 
-    private handleSocketOpen = async () => {
+    private handleSocketOpen = (resolve, reject) => async () => {
```

Comment:
```suggestion
    private handleSocketOpen = async () => {
```
change to that please

Reply:
this will break it